### DISCUSSION TITLE
`vec` should be renamed to `col`, and `arrow` should be renamed to `vec`

### DISCUSSION THREAD

**Discussion by K4LCIFER** [typst/typst#3538] [Ideas]
Currently, `vec` is used for creating column vectors, but I think it should instead be used for what `arrow` does. People coming from latex would be used to `vec` when creating vector variables, and it is also less keystrokes/less verbose when writing math (some equations would become rather cumbersome to write, if one has to be using `arrow`, instead of `vec`). I would then propose that a new function `col` (for column vector) be introduced to take the place of what `vec` was. In the same vein, a function `row` could be used for a row vector.

**Comment by Enivex** [comment]
There are other ways of denoting vectors than an arrow. Boldface being a big one. Moreover, there can be other uses of an arrow than denoting vectors.

The current name doesn't carry semantic meaning, and I view that as a good thing. You can always define your own name for it if you so desire.

I'm less opposed to renaming vec to col. Adding row seems pointless though, as you could equally well just use mat.

**Reply by laurmaedje** [reply]
Generally, I think functions _should_ carry semantic meaning. All these different ways to denote a vector would ideally be produced via show rules in my opinion. However, I don't really like col for that same reason.

**Reply by Enivex** [reply]
> Generally, I think functions _should_ carry semantic meaning. All these different ways to denote a vector would ideally be produced via show rules in my opinion. However, I don't really like col for that same reason.

I don't disagree that as a user it's better to use functions with semantic meaning, but I'd prefer using let bindings over show.

Both because the user can choose their own semantic name for their own use case, and because it avoids the headache of having to revoke the show rule (when that becomes a thing) to use the original function.

**Comment by HPDell** [comment]
From my view, a vector is a column vector by default. So it doesn't matter to use the name `vec`. But I would prefer a kind of "smart vector". In the inline mode, it shows elements horizontally and append a transform mark at the end. In the display mode, it shows elements vertically.

For example,

```typst
$vec(a_1,a_2,a_3)$
```

will looks like: $\left(a_1,a_2,a_3\right)^\mathrm{T}$.

But

```typst
$ vec(a_1,a_2,a_3) $
```

will looks like:

![image](https://github.com/typst/typst/assets/17842185/ddb0d87f-e85e-44ed-a077-10fa1658e2eb)

**Reply by AxiomOfChoices** [reply]
Personally, I think this type of conditional rendering is best left for packages, and you currently already implement in typst the behavior you want.

```
#show math.equation.where(block: false): it => {
  show math.vec: v => {
    [$(#v.children.slice(1).fold([#v.children.at(0)],(a,b) => $#a,#b$))^T$]
  }
  it
}


Inline:
$vec(a_1,a_2,a_3)$

Block:
$
  vec(a_2,a_2,a_3)
$
```

![image](https://github.com/typst/typst/assets/14252368/5f704991-f566-4ce0-9bf3-1c627357038d)

**Reply by HPDell** [reply]
Thank you so much! I nearly forgot the `where` condition can be used on `equation`.
